plugins/out_stackdriver/stackdriver.c

```diff
@@ -2758,20 +2765,25 @@ static void cb_stackdriver_flush(struct flb_event_chunk *event_chunk,
         FLB_OUTPUT_RETURN(FLB_RETRY);
     }
 
-    /* Reformat msgpack to stackdriver JSON payload */
-    payload_buf = stackdriver_format(ctx,
-                                     event_chunk->total_events,
-                                     event_chunk->tag, flb_sds_len(event_chunk->tag),
-                                     event_chunk->data, event_chunk->size);
-    if (!payload_buf) {
+    if (ctx->test_log_entry_format) {
+        printf("%s", payload_buf);
+        flb_sds_destroy(payload_buf);
+        FLB_OUTPUT_RETURN(FLB_OK);
+    }
+
+    /* Get upstream connection */
+    u_conn = flb_upstream_conn_get(ctx->u);
+    if (!u_conn) {
 #ifdef FLB_HAVE_METRICS
         cmt_counter_inc(ctx->cmt_failed_requests,
                         ts, 1, (char *[]) {name});
 
         /* OLD api */
         flb_metrics_sum(FLB_STACKDRIVER_FAILED_REQUESTS, 1, ctx->ins->metrics);
+
+        update_retry_metric(ctx, event_chunk, ts, STACKDRIVER_NET_ERROR);
```

Comment:
I think you also need to delete this from line 2763 — there's no request to retry when you're unable to allocate the payload.